R/exposure_pie_chart.R

```diff
@@ -0,0 +1,103 @@
+#' Create an interactive exposure pie chart in an htmlwidget
+#'
+#' @param .data a exposure pie data frame
+#' @param width width of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param height height of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param ... other options passed on to r2d3::r2d3() (see details)
+#'
+#' @description
+#' The `exposure_pie_chart` function creates an interactive exposure pie chart in an htmlwidget
+#'
+#' @md
+#' @export
+
+exposure_pie_chart <-
+  function(.data, width = NULL, height = NULL, ...) {
+    .data <- export_data_utf8(.data)
+
+    dependencies <-
+      list(
+        system.file("js/pie_exploded.js", package = "r2dii.interactive"),
+        system.file("js/d3-array.v2.min.js", package = "r2dii.interactive"),
+        system.file("js/d3-scale.v3.min.js", package = "r2dii.interactive"),
+        system.file("js/d3-scale-chromatic.v1.min.js", package = "r2dii.interactive"),
+        system.file("js/jquery-3.5.1.js", package = "r2dii.interactive"),
+        system.file("js/text_dropdown_jiggle.js", package = "r2dii.interactive"),
+        system.file("css/2dii_gitbook_style.css", package = "r2dii.interactive"),
+        system.file("css/hide_styles.css", package = "r2dii.interactive")
+        )
+
+    op <- options(r2d3.shadow = FALSE)
+    on.exit(options(op), add = TRUE)
+
+    r2d3::r2d3(
+      data = .data,
+      script = system.file("render_exposure_pie.js", package = "r2dii.interactive"),
+      dependencies = dependencies,
+      d3_version = 4,
+      width = width,
+      height = height,
+      container = "div",
+      ...
+    )
+  }
+
+
+#' Convert raw data into a exposure pie data frame
+#'
+#' @param audit_file audit_file
+#' @param investor_name investor_name
+#' @param portfolio_name portfolio_name
+#' @param twodi_sectors twodi_sectors
+#' @param dataframe_translations dataframe_translations
+#' @param language_select two letter code for language (single string; default = "EN")
+#'
+#' @description
+#' The `as_exposure_pie_data` function converts raw data into a exposure pie data frame
+#'
+#' @import dplyr
```

Comment:
I see `@import dplyr` also at https://github.com/2DegreesInvesting/r2dii.interactive/blob/cfe4d506624dd36eb4d0ab3a9901621c0803c636/R/utils.R#L46 and I see `dplyr::` is sometimes used. 

This is no big deal but in my experience it's best to do all imports in a single place, e.g. R/imports.R. That is a
predictable place where to look for to learn what packages and functions are already being imported.

Reply:
I disagree. I like to have the import commands just above the functions that need them. Same argument as for keeping the roxygen comments just above.